src/Components/TestExplorer.fs

```diff
@@ -1180,17 +1193,32 @@ module Interactions =
 
         let testToFilterExpression (test: TestItem) =
             let fullTestName = TestItem.getFullName test.id
-            let fullName = escapeFilterExpression fullTestName
+            let escapedTestName = escapeFilterExpression fullTestName
 
-            if fullName.Contains(" ") && test.TestFramework = TestFrameworkId.NUnit then
+            if escapedTestName.Contains(" ") && test.TestFramework = TestFrameworkId.NUnit then
                 // workaround for https://github.com/nunit/nunit3-vs-adapter/issues/876
                 // Potentially we are going to run multiple tests that match this filter
-                let testPart = fullName.Split(' ').[0]
+                let testPart = escapedTestName.Split(' ').[0]
                 $"(FullyQualifiedName~{testPart})"
+            else if test.TestFramework = TestFrameworkId.XUnit then
```

Comment:
This is the core of the change. It customizes the filter expression based on test framework.